server/src/infra/repositories/media.repository.ts

```diff
@@ -23,10 +25,25 @@ export class MediaRepository implements IMediaRepository {
   }
 
   async resize(input: string | Buffer, output: string, options: ResizeOptions): Promise<void> {
-    await sharp(input, { failOn: 'none' })
+    let colorProfile = options.colorspace;
+    if (options.colorspace !== Colorspace.SRGB) {
+      try {
+        const { space } = await sharp(input).metadata();
+        // if the image is already in srgb, keep it that way
+        if (space && space === Colorspace.SRGB) {
```

Comment:
I also don't get what's special about sRGB? If the original space is AdobeRGB and the target is P3 for instance, then why is it okay to use P3 for that and not sRGB?

Reply:
It doesn't make sense to convert sRGB content to P3. The reason to use P3 is because of its wider color gamut, but content in sRGB doesn't stand to gain anything from this and can only lose due to precision loss during conversion. 

AdobeRGB isn't a colorspace in libvips. It would show up as generic rgb16 or similar. P3 is a better choice for this and other content because there's a higher chance of it supporting the colors in the source.